plugin/kubernetai/kubernetai.go

```diff
@@ -83,14 +87,25 @@ func (k8i Kubernetai) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns
 
 // AutoPath routes AutoPath requests to the authoritative kubernetes.
 func (k8i Kubernetai) AutoPath(state request.Request) []string {
+	var searchPath []string
 	for _, k := range k8i.Kubernetes {
-		zone := plugin.Zones(k.Zones).Matches(state.Name())
-		if zone == "" {
-			continue
+		zones := make([]string, 0, len(k.Zones)*2)
+		zones = append(zones, k.Zones...)
+		for _, z := range k.Zones {
```

Comment:
This section isn't quite right.  It needs to build the full search list for each unique zone...  so for 2 unique zones `cluster1.local.` and `cluster2.local.` we should generate...

~~~
<client-pod-namespace>.svc.cluster1.local.
svc.cluster1.local.
cluster1.local.
<client-pod-namespace>.svc.cluster2.local.
svc.cluster2.local.
cluster2.local.
<cluster-search-domains>
~~~

<client-pod-namespace> = the namespace of the client pod
<cluster-search-domains> = the search domains on the k8s host nodes. 

or breadth-first ...

~~~
<client-pod-namespace>.svc.cluster1.local.
<client-pod-namespace>.svc.cluster2.local.
svc.cluster1.local.
svc.cluster2.local.
cluster1.local.
cluster2.local.
<cluster-search-domains>
~~~